src/TurtleParserMain.cpp

```diff
@@ -2,36 +2,166 @@
+               << ". This should never happen, terminating" << std::endl;
+    LOG(ERROR) << "Please specify a valid file format" << std::endl;
     exit(1);
   }
+}
+
+// _______________________________________________________________________________________________________________
+void printUsage(char* execName) {
+  std::ios coutState(nullptr);
+  coutState.copyfmt(cout);
+  cout << std::setfill(' ') << std::left;
+
+  cout << "  " << std::setw(20) << "F, file-format" << std::setw(1) << "    "
+       << " Specify format of the input file. Must be one of "
+          "[tsv|nt|ttl|mmap]."
+       << " " << std::setw(36)
+       << "If not set, we will try to deduce from the filename" << endl
+       << " " << std::setw(36)
+       << "(mmap assumes an on-disk turtle file that can be mmapped to memory)"
+       << endl;
+  cout << "  " << std::setw(20) << "i, input-file" << std::setw(1) << "    "
+       << " The file to be parsed from. If omitted, we will read from stdin"
+       << endl;
+  cout << "  " << std::setw(20) << "o, output-file" << std::setw(1) << "    "
+       << " The NTriples file to be Written to. If omitted, we will write to "
+          "stdout"
+       << endl;
+  cout.copyfmt(coutState);
+}
+
+// ________________________________________________________________________
+int main(int argc, char** argv) {
+  // we possibly write to stdout to pipe it somewhere else, so redirect all
+  // logging output to std::err
+  ad_utility::setGlobalLogginStream(&std::cerr);
+  struct option options[] = {{"help", no_argument, NULL, 'h'},
+                             {"file-format", required_argument, NULL, 'F'},
+                             {"input-file", required_argument, NULL, 'i'},
+                             {"output-file", required_argument, NULL, 'o'},
+                             {NULL, 0, NULL, 0}};
+
+  string inputFile, outputFile, fileFormat;
+  while (true) {
+    int c = getopt_long(argc, argv, "F:i:o:h", options, nullptr);
+    if (c == -1) {
+      break;
+    }
+    switch (c) {
+      case 'h':
+        printUsage(argv[0]);
+        return 0;
+        break;
+      case 'i':
+        inputFile = optarg;
+        break;
+      case 'o':
+        outputFile = optarg;
+        break;
+      case 'F':
+        fileFormat = optarg;
+        break;
+      default:
+        cout << endl
+             << "! ERROR in processing options (getopt returned '" << c
+             << "' = 0x" << std::setbase(16) << c << ")" << endl
+             << endl;
+        printUsage(argv[0]);
+        exit(1);
+    }
+  }
+
+  if (fileFormat.empty()) {
+    if (ad_utility::endsWith(inputFile, ".tsv")) {
+      fileFormat = "tsv";
+    } else if (ad_utility::endsWith(inputFile, ".nt")) {
+      fileFormat = "nt";
+    } else if (ad_utility::endsWith(inputFile, ".ttl")) {
+      fileFormat = "ttl";
+    } else if (ad_utility::endsWith(inputFile, ".ttl.bz2")) {
+      fileFormat = "bzip2";
```

Comment:
here it's `bzip2` above it was `mmap` and I don't see where we still link with the `bz2` lib

Reply:
fixed